File: erpnext/assets/report/asset_by_agency/asset_by_agency.js
// Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Asset by Agency"] = {
	"filters": [
		{
            "fieldname": "company",
            "label": __("Company"),
            "fieldtype": "Link",
            "options": "Company",
            "reqd": 0 // Set to 1 if you want to make this filter mandatory
        }
	]
};

File: erpnext/assets/report/asset_by_agency/asset_by_agency.py
# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt


from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns, data = get_column(), get_data(filters)
	return columns, data

def get_column():
    columns = [
		("Agency") + ":Link/eNote:100",
		("Assets") + ":Data:200",
    ]
    return columns

def get_data(filters):
    conditions = get_condition(filters)
    return frappe.db.sql("""
        select company,count(name) from `tabAsset` where {conditions} group by company ;
		""".format(conditions=conditions))
   
    
    
def get_condition(filters):
    conds = "1=1 "
    if filters.company:
        conds += "and company='{}'".format(filters.company)
    return conds

